mongodbatlas/resource_mongodbatlas_api_key.go

```diff
@@ -92,8 +92,12 @@ func resourceMongoDBAtlasAPIKeyRead(ctx context.Context, d *schema.ResourceData,
 	orgID := ids["org_id"]
 	apiKeyID := ids["api_key_id"]
 
-	apiKey, _, err := conn.APIKeys.Get(ctx, orgID, apiKeyID)
+	apiKey, resp, err := conn.APIKeys.Get(ctx, orgID, apiKeyID)
 	if err != nil {
+		if resp != nil && resp.StatusCode == http.StatusNotFound || resp.StatusCode == http.StatusBadRequest {
+			d.SetId("")
+			return nil
```

Comment:
why do we also consider `StatusBadRequest`? Shouldn't we only consider StatusNotFound?

Reply:
If they deleted API key and it tries to find it comes back with bad request so had to catch that scenario